src/Request.ts

```diff
@@ -0,0 +1,320 @@
+					}
+				});
+			}),
+			response: response
+		};
+	}
+
+	_renderFilterParams(filter: Filter) {
+		// summary:
+		//		Constructs filter-related params to be inserted into the query string
+		// returns: String
+		//		Filter-related params to be inserted in the query string
+		var type = filter.type;
+		var args = filter.args;
+		if (!type) {
+			return [''];
+		}
+		if (type === 'string') {
+			return [args[0]];
+		}
+		if (type === 'and' || type === 'or') {
+			return [filter.args.map(function (arg) {
+				// render each of the arguments to and or or, then combine by the right operator
+				var renderedArg = this._renderFilterParams(arg);
+				return ((arg.type === 'and' || arg.type === 'or') && arg.type !== type) ?
+					// need to observe precedence in the case of changing combination operators
+					'(' + renderedArg + ')' : renderedArg;
+			}, this).join(type === 'and' ? '&' : '|')];
+		}
+		var target = args[1];
+		if (target) {
+			if(target._renderUrl) {
+				// detected nested query, and render the url inside as an argument
+				target = '(' + target._renderUrl() + ')';
+			} else if (target instanceof Array) {
+				target = '(' + target + ')';
+			}
+		}
+		return [encodeURIComponent(args[0]) + '=' + (type === 'eq' ? '' : type + '=') + encodeURIComponent(target)];
+	}
+
+	_renderSortParams(sort: dstore.SortOption[]) {
+		// summary:
+		//		Constructs sort-related params to be inserted in the query string
+		// returns: String
+		//		Sort-related params to be inserted in the query string
+		var sortString = sort.map(function (sortOption: dstore.SortOption) {
+			var prefix = sortOption.descending ? this.descendingPrefix : this.ascendingPrefix;
+			return prefix + encodeURIComponent(sortOption.property);
+		}, this);
+
+		var params: string[] = [];
+		if (sortString) {
+			params.push(this.sortParam
+				? encodeURIComponent(this.sortParam) + '=' + sortString
+				: 'sort(' + sortString + ')'
+			);
+		}
+		return params;
+	}
+
+	_renderRangeParams(start: number, end: number) {
+		// summary:
+		//		Constructs range-related params to be inserted in the query string
+		// returns: String
+		//		Range-related params to be inserted in the query string
+		var params: string[] = [];
+		if (this.rangeStartParam) {
+			params.push(
+				this.rangeStartParam + '=' + start,
+				this.rangeCountParam + '=' + (end - start)
+			);
+		} else {
+			params.push('limit(' + (end - start) + (start ? (',' + start) : '') + ')');
+		}
+		return params;
+	}
+
+	_renderSelectParams(properties: string) {
+		// summary:
+		//		Constructs select-related params to be inserted in the query string
+		// returns: String
+		//		Select-related params to be inserted in the query string
+		var params: string[] = [];
+		if (this.selectParam) {
+			params.push(this.selectParam + '=' + properties);
+		} else {
+			params.push('select(' + properties + ')');
+		}
+		return params;
+	}
+
+	_renderQueryParams() {
+		var queryParams: Array<string| {}> = [];
```

Comment:
`Array<string | {}>` is essentially the same as `Array<any>` because of the `{}`